tony-core/src/main/java/com/linkedin/tony/models/JobLog.java

```diff
@@ -0,0 +1,174 @@
+    }
+    return wrapper;
+  }
+
+  /**
+   *
+   * @param e {@link Event}
+   * @return Pair contains node host address and container id if desired  event type is there
+   * return null is other events there
+   */
+  private static Pair<String, String> processEventForNodeIdContainerId(Event e) {
+    Pair<String, String> nodeIDContainerID = null;
+    switch (e.getType()) {
+      case APPLICATION_INITED:
+        nodeIDContainerID =
+            Pair.of(((ApplicationInited) e.getEvent()).getHost(), ((ApplicationInited) e.getEvent()).getContainerID());
+        break;
+      case TASK_STARTED:
+        nodeIDContainerID =
+            Pair.of(((TaskStarted) e.getEvent()).getHost(), ((TaskStarted) e.getEvent()).getContainerID());
+        break;
+      default:
+        LOG.debug(" Event type doesnt have node id  and container id information " + e.getType().name());
+        break;
+    }
+    return nodeIDContainerID;
+  }
+
+  /**
+   * @param nodeIdContainerId: Node Id and container Id information
+   * @param yarnConfiguration Yarnconfiguration provided to TonY
+   * @param userName username who launched the job
+   * @return LogLink , it will be NA for TASK_FINISHED and APPLICATION_FINISHED event
+   * APPLICATION_INITED, TASK_STARTED value which will be jobhistory URL to see the container logs
+   */
+  private static String createLogLink(Pair<String, String> nodeIdContainerId, YarnConfiguration yarnConfiguration,
+      String userName) {
+    String logLink = DEFAULT_VALUE_OF_CONTAINER_LOG_LINK;
+    if (isParameterValid(userName)) {
+      LOG.debug(" Valid values in yarn configuration and username");
+      Pair<String, String> jobHistoryNodeMgrAdd = getJobHistoryNodeMgrAdd(yarnConfiguration);
+      if (isParameterValid(jobHistoryNodeMgrAdd)) {
+        LOG.debug("Create container log URL with following information " + " " + jobHistoryNodeMgrAdd.getLeft() + " "
+            + jobHistoryNodeMgrAdd.getRight() + " " + nodeIdContainerId.getLeft() + " " + nodeIdContainerId.getRight());
+        logLink = processForLogLink(jobHistoryNodeMgrAdd, nodeIdContainerId, userName);
+      }
+    }
+    return logLink;
+  }
+
+  /**
+   * @param yarnConfiguration {@link YarnConfiguration} yarnconfiguration provided to TonY
+   * @return Pair contains mapreduce.jobhistory.webapp.address if able to get information from
+   * yarnconf ,other will return null
+   */
+  private static Pair<String, String> getJobHistoryNodeMgrAdd(YarnConfiguration yarnConfiguration) {
+    Pair<String, String> jobHistoryNodeMgrAddress = null;
+    if (isParameterValid(yarnConfiguration)) {
+      String mapReduceJobHistoryWebAddress = yarnConfiguration.get(MAPREDUCE_JOBHISTORY_WEBAPP_ADDRESS);
+      String yarnNodeManagerAddress = yarnConfiguration.get(YARN_NODEMANAGER_ADDRESS);
+      if (isParameterValid(mapReduceJobHistoryWebAddress, yarnNodeManagerAddress)) {
```

Comment:
not getting this portion, I think you are trying to assemble the link to logs, why do you need MR web addess? YARN NM address should be enough

Reply:
Removed, changed the logic